### PR TITLE
fix issue 1053 non-string args to path.join in 0.10.x

### PR DESCRIPTION
**Pull request by isao**

### CODE REVIEW

File: lib/app/autoload/store.server.js [hunk 1]
```diff
@@ -1692,11 +1692,11 @@ YUI.add('mojito-resource-store', function(Y, NAME) {
 
             switch (info.pkg.yahoo.mojito.type) {
             case 'bundle':
-                dir = this._libs.path.join(info.dir, info.pkg.yahoo.mojito.location);
+                dir = this._libs.path.join(info.dir, info.pkg.yahoo.mojito.location || '');
```
Comment (caridy) [inline lib/app/autoload/store.server.js:1695]: when this location member can be undefined?

File: lib/app/autoload/store.server.js [hunk 2]
```diff
@@ -709,7 +709,7 @@ YUI.add('mojito-resource-store', function(Y, NAME) {
                 if (res.type === 'view') {
                     template = {
                         'content-path': (env === 'client' ?
-                                this._libs.path.join(this._appConfigStatic.pathToRoot || '', res.url) :
+                                this._libs.path.join(this._appConfigStatic.pathToRoot || '', res.url || '') :
```
Comment (caridy) [inline lib/app/autoload/store.server.js]: when the res.url will be undefined/null?

File: lib/app/autoload/store.server.js (hunk 1 above)
Comment (isao) [inline lib/app/autoload/store.server.js:1695]: when a user has a bundle without a location property. It's defined [here](https://github.com/yahoo/mojito/blob/develop/tests/fixtures/packages/node_modules/a/package.json) but not [here](https://github.com/yahoo/mojito/blob/develop/tests/fixtures/packages/node_modules/b/package.json) for example.

File: lib/app/autoload/store.server.js (hunk 2 above)
Comment (isao) [inline lib/app/autoload/store.server.js]: during [this test](https://github.com/yahoo/mojito/blob/develop/tests/unit/lib/app/autoload/test-store.server.js#L500) for [this fixture](https://github.com/yahoo/mojito/tree/develop/tests/fixtures/packages).

Comment (drewfish) [inline lib/app/autoload/store.server.js]: I still think adding the real addon-rs-url in the descriptor is a cleaner fix (if it works).

Comment (caridy) [inline lib/app/autoload/store.server.js]: me too!

Comment (isao) [inline lib/app/autoload/store.server.js]: So res.url is always defined? I thought it wasn't in some cases on the client i.e. shared mojits. If no res.url is valid, then 'content-path' should not be defined with a unusable value. If the test exposes a case where res.url should be defined but isn't, then I think we should open a separate issue.

Comment (caridy) [inline lib/app/autoload/store.server.js]: Yes, we remove that exception a while ago, and we expect that url is always defined.

File: lib/app/autoload/store.server.js (hunk 1 above)
Comment (caridy) [inline lib/app/autoload/store.server.js:1695]: So, location is not produced by the store? it just comes from the pkg information as optional then? if that's the case, we are good!

**Comment by isao** [comment]
thanks for the feedback @caridy @drewfish. the root of the undefined res.url issue was test mock, not store code. unit tests pass on 0.10.2 mac.

**Comment by drewfish** [comment]
+1

**Pull request merged**